skyportal/models.py

```diff
@@ -304,6 +304,32 @@ def target(self):
         coord = ap_coord.SkyCoord(self.ra, self.dec, unit='deg')
         return astroplan.FixedTarget(name=self.id, coord=coord)
 
+    @hybrid_property
+    def gal_lat(self):
+        """Get the galactic latitute of this object"""
+        coord = ap_coord.SkyCoord(self.ra, self.dec, unit="deg")
+        return coord.galactic.b.deg
+
+    @gal_lat.expression
+    def gal_lat(cls):
+        ra_array = DBSession.query(cls.ra).all()
+        dec_array = DBSession.query(cls.dec).all()
+        coords = ap_coord.SkyCoord(ra_array, dec_array, unit="deg")
+        return sa.orm.query.Query(coords.galactic.b.deg).label("gal_lat")
+
+    @hybrid_property
+    def gal_lon(self):
+        """Get the galactic longitude of this object"""
+        coord = ap_coord.SkyCoord(self.ra, self.dec, unit="deg")
+        return coord.galactic.l.deg
+
+    @gal_lon.expression
+    def gal_lon(cls):
+        ra_array = DBSession.query(cls.ra).all()
+        dec_array = DBSession.query(cls.dec).all()
+        coords = ap_coord.SkyCoord(ra_array, dec_array, unit="deg")
+        return sa.orm.query.Query(coords.galactic.l.deg).label("gal_lon")
```

Comment:
I believe this is will error or silently fail as `coords.galactic.l.deg` is not an SQLalchemy entity. @guynir42 Can you add a test for this?

Reply:
in the preceding lines, the code can pull the entire ra, dec columns of the table (potentially billions of rows) into memory, convert them to to galactic coords client side, then (if line 331 does not throw an exception) push the results back to the database as a literal array. If the conversion is happening client side anyway, maybe best to implement this as a straight up `@property` instead of a `@hybrid_property` (i.e., delete this method and re-implement the corresponding getter as `@property`)?

Reply:
First off you are right on the first comment: I can just as easily return an array and it seems to work. I am not sure what kind of test to do but the coordinates show up on all the relevant pages. The only question I am now left with is how it worked to begin with...?

As for your second comment, I think that since there are two "getters" in the hybrid property, it can choose which ever one is more useful, for pulling one item at a time or to pull the entire column. I have no idea what is more efficient or if SQLAlchemy is smart enough to do the right thing. So whatever you think is best to do.